src/name.ts

```diff
@@ -0,0 +1,298 @@
+import type { Faker } from '.';
+
+export class Name {
+  constructor(private readonly faker: Faker) {
+    // Bind `this` so namespaced is working correctly
+    for (const name of Object.getOwnPropertyNames(Name.prototype)) {
+      if (name === 'constructor' || typeof this[name] !== 'function') {
+        continue;
+      }
+      this[name] = this[name].bind(this);
+    }
+  }
+
+  /**
+   * firstName
+   *
+   * @method faker.name.firstName
+   * @param {mixed} gender
+   * @memberof faker.name
+   */
+  firstName(gender?: string | number): string {
+    if (
+      typeof this.faker.definitions.name.male_first_name !== 'undefined' &&
+      typeof this.faker.definitions.name.female_first_name !== 'undefined'
+    ) {
+      // some locale datasets ( like ru ) have first_name split by gender. since the name.first_name field does not exist in these datasets,
+      // we must randomly pick a name from either gender array so faker.name.firstName will return the correct locale data ( and not fallback )
+
+      if (typeof gender === 'string') {
+        if (gender.toLowerCase() === 'male') {
+          gender = 0;
+        } else if (gender.toLowerCase() === 'female') {
+          gender = 1;
+        }
+      }
+
+      if (typeof gender !== 'number') {
+        if (typeof this.faker.definitions.name.first_name === 'undefined') {
+          gender = this.faker.datatype.number(1);
+        } else {
+          //Fall back to non-gendered names if they exist and gender wasn't specified
```

Comment:
Would be cool if we already format a bit such comments

```suggestion
          // Fall back to non-gendered names if they exist and gender wasn't specified
```